fvtest/compilertriltest/ShiftAndRotateTest.cpp

```diff
@@ -1017,14 +1017,38 @@ std::vector<T> test_masks();
 
 static uint64_t mask64Values[] =
     {
-    0x0000000000000000,
-    0xffffffffffffffff,
+    0x0,
```

Comment:
Is there a reason to be adding this many new 64-bit mask values? Are these testing corner cases that weren't covered by the previous values? The mask-then-shift tests are using the cross-product of three different sets of test values (one generic value, one mask, and one shift), so it's really easy to get combinatorial explosion by adding too many here...

Reply:
These were all of the corner cases that were accounted for in the removal of the `LongAndAsRotate` test. I guess that these corner cases were only technically tested previously with a shift value of zero.

Reply:
The new mask values come from [1] for which there is a comment explaining all the different cases we are trying to cover. If the newly added masks do not cover additional cases, then yes I agree we don't need to add them. @AidanHa can you go through the existing masks before your change and identify where we are adding coverage with all the new mask values? Was there a case the original mask values were not capturing?

Perhaps there is redundancy to some level. I'm not sure.

[1] https://github.com/eclipse/omr/blob/d309a6eaafd43ca1040175a09c0a68f30fe4357e/fvtest/compilertriltest/LongAndAsRotateTest.cpp#L32-L61

Reply:
Most of the newly added mask values came from removing the `LongAndAsRotate` test. Now that We've decided to leave that test alone for now, we can remove a lot of these values. I'll have a push soon that will remove all of the redundant values.

Reply:
The ones that will be added were edge cases captured while testing this feature. I've also added some cases where the new feature should NOT use the optimization.

Reply:
@aviansie-ben I think the number of new cases has been reduced. Could you please re-review? Let me know which threads can be resolved.